ovs/lib/helpers/generic/scrubber.py

```diff
@@ -0,0 +1,1066 @@
+    def _format_message(self, message):
+        """
+        Formats a message with the set _log property in front (to remove redundancy)
+        :param message: The message to format
+        :return: Newly formatted message
+        """
+        if self._log is None:
+            raise ValueError('_log property has no value. Nothing to format with')
+        return '{0} - {1}'.format(self._log, message)
+
+
+class StackWorkHandler(ScrubShared):
+    """
+    Handles generation, unregistering and saving of vpool stack work
+    """
+
+    def __init__(self, job_id, vpool, vdisks, worker_contexts):
+        """
+        Initialize
+        :param vpool: vPool to generate work for
+        :type vpool: ovs.dal.hybrids.vpool.VPool
+        :param vdisks: vDisks to include in the work
+        :type vdisks: list
+        :param worker_contexts: Contexts about the workers
+        :type worker_contexts: dict
+        """
+        super(StackWorkHandler, self).__init__(job_id)
+
+        self.vpool = vpool
+        self.vdisks = vdisks
+        self.worker_contexts = worker_contexts
+
+        self._key = self._SCRUB_VDISK_KEY.format(self.vpool.name)  # Key to register items under
+        self._log = 'Scrubber {0} - vPool {1}'.format(self.job_id, self.vpool.name)
+
+    def generate_save_scrub_work(self):
+        """
+        Generates applicable scrub work and saves the scrub work consistently
+        :return: Queue of work items
+        """
+        work_queue = self._generate_save_work()
+        self._logger.info(self._format_message('Following items will be scrubbed: {0}'.format(list(work_queue.queue))))
+        return work_queue
+
+    def _generate_scrub_work(self):
+        """
+        Generates scrub work to be done
+        :return: Queue of work items, value to store in persistent, the current expected value
+        :rtype: tuple(Queue, list, list)
+        """
+        # Fetch data
+        relevant_work_items, fetched_work_items = self._get_pending_scrub_work()
+        registered_vdisks = [item['vdisk_guid'] for item in relevant_work_items]
+        work_queue = Queue()
+        for vd in self.vdisks:
+            logging_start_vd = '{0} - vDisk {1} with guid {2}'.format(self._log, vd.name, vd.guid)
+            if vd.guid in registered_vdisks:
+                self._logger.info('{0} has already been registered to get scrubbed, not queueing again'.format(logging_start_vd))
+                continue
+            if vd.is_vtemplate is True:
+                self._logger.info('{0} is a template, not scrubbing'.format(logging_start_vd))
+                continue
+            vd.invalidate_dynamics('storagedriver_id')
+            if not vd.storagedriver_id:
+                self._logger.warning('{0} no StorageDriver ID found'.format(logging_start_vd))
+                continue
+            work_queue.put(vd.guid)
+        total_items = list(itertools.chain(relevant_work_items, (self._wrap_data(item) for item in work_queue.queue)))
```

Comment:
Since this is a simple add operation, i'd prefer a '+' for the sake of comprehensibility